schema/resolvers.test.js

```diff
@@ -0,0 +1,161 @@
+const repositories = require("../repositories");
+const db = require("../db");
+const executeQuery = require("../tests/utils/executeQuery");
+const {
+  createQuestionnaire,
+  createAnswer,
+  createOtherAnswer,
+  deleteOtherAnswer
+} = require("../tests/utils/graphql");
+
+const ctx = { repositories };
+
+const createNewQuestionnaire = async () => {
+  const input = {
+    title: "Test Questionnaire",
+    description: "Questionnaire created by integration test.",
+    theme: "default",
+    legalBasis: "Voluntary",
+    navigation: false,
+    surveyId: "001",
+    summary: true,
+    createdBy: "Integration test"
+  };
+
+  const result = await executeQuery(createQuestionnaire, { input }, ctx);
+  return result.data.createQuestionnaire;
+};
+
+const createNewAnswer = async (pageId, type) => {
+  const input = {
+    description: "",
+    guidance: "",
+    label: `${type} answer`,
+    qCode: null,
+    type: `${type}`,
+    mandatory: false,
+    questionPageId: pageId
+  };
+
+  const result = await executeQuery(createAnswer, { input }, ctx);
+  return result.data.createAnswer;
+};
+
+const createNewOtherAnswer = async answer => {
+  const result = await executeQuery(
+    createOtherAnswer,
+    { input: { parentAnswerId: answer.id } },
+    ctx
+  );
+  return result.data.createOtherAnswer;
+};
+
+const createThenDeleteOtherAnswer = async (pageId, type) => {
+  let answer = await createNewAnswer(pageId, type);
+  answer = await createNewOtherAnswer(answer);
+
+  const result = await executeQuery(
+    deleteOtherAnswer,
+    {
+      input: {
+        parentAnswerId: answer.id
+      }
+    },
+    ctx
+  );
+
+  return result.data.deleteOtherAnswer;
+};
+
+describe("resolvers", () => {
+  let questionnaire;
+  let sections;
+  let pages;
+
+  beforeAll(() => db.migrate.latest());
+  afterAll(() => db.migrate.rollback().then(() => db.destroy()));
+  afterEach(() => db("Questionnaires").delete());
+
+  beforeEach(async () => {
+    questionnaire = await createNewQuestionnaire();
+    sections = questionnaire.sections;
+    pages = sections[0].pages;
+  });
+
+  it("should create other answer for Checkbox answers", async done => {
+    let checkboxAnswer = await createNewAnswer(pages[0].id, "Checkbox");
+    expect(checkboxAnswer.otherAnswer).toBeNull();
+
+    checkboxAnswer = await createNewOtherAnswer(checkboxAnswer);
+
+    expect(checkboxAnswer.otherAnswer).not.toBeNull();
+    expect(checkboxAnswer.otherAnswer).toMatchObject({
+      type: "TextField"
+    });
+    done();
```

Comment:
you shouldn't need to call `done()` from any of these. All `async` functions return a promise, and jest automatically waits for promises to resolve